stacks-common/src/util/macros.rs

```diff
@@ -88,30 +88,50 @@ macro_rules! define_named_enum {
 ///  and EnumType.get_name() for free.
 #[macro_export]
 macro_rules! define_versioned_named_enum {
-    ($Name:ident($VerType:ty) { $($Variant:ident($VarName:literal, $Version:expr),)* }) =>
-    {
+    ($Name:ident($VerType:ty) { $($Variant:ident($VarName:literal, $MinVersion:expr)),* $(,)* }) => {
+        $crate::define_versioned_named_enum_internal!($Name($VerType) {
+            $($Variant($VarName, $MinVersion, None)),*
+        });
+    };
+}
+#[macro_export]
+macro_rules! define_versioned_named_enum_with_max {
+    ($Name:ident($VerType:ty) { $($Variant:ident($VarName:literal, $MinVersion:expr, $MaxVersion:expr)),* $(,)* }) => {
+        $crate::define_versioned_named_enum_internal!($Name($VerType) {
+            $($Variant($VarName, $MinVersion, $MaxVersion)),*
+        });
+    };
+}
+
+// An internal macro that does the actual enum definition
+#[macro_export]
+macro_rules! define_versioned_named_enum_internal {
+    ($Name:ident($VerType:ty) { $($Variant:ident($VarName:literal, $MinVersion:expr, $MaxVersion:expr)),* $(,)* }) => {
         #[derive(::serde::Serialize, ::serde::Deserialize, Debug, Hash, PartialEq, Eq, Copy, Clone)]
         pub enum $Name {
             $($Variant),*,
         }
+
         impl $Name {
             pub const ALL: &'static [$Name] = &[$($Name::$Variant),*];
             pub const ALL_NAMES: &'static [&'static str] = &[$($VarName),*];
 
             pub fn lookup_by_name(name: &str) -> Option<Self> {
                 match name {
-                    $(
-                        $VarName => Some($Name::$Variant),
-                    )*
-                    _ => None
+                    $($VarName => Some($Name::$Variant),)*
+                    _ => None,
                 }
             }
 
-            pub fn get_version(&self) -> $VerType {
+            pub fn get_min_version(&self) -> $VerType {
                 match self {
-                    $(
-                        $Name::$Variant => $Version,
-                    )*
+                    $(Self::$Variant => $MinVersion,)*
+                }
+            }
+
+            pub fn get_max_version(&self) -> Option<$VerType> {
+                match self {
+                    $(Self::$Variant => $MaxVersion,)*
```

Comment:
Can you add rustdocs to the functions e.g., `get_min_version` returns the first Clarity version in which `self` is defined, `get_max_version` returns `Some` for the last Clarity version in which `self` is defined, or `None` if `self` is defined in every Clarity version after `get_min_version()`.

Reply:
Added in aa832234791c8d902c44618dabbf0f5d48a7546d.